packages/trader/src/Modules/Trading/Components/Elements/purchase-button.jsx

```diff
@@ -74,19 +70,19 @@ const PurchaseButton = ({
 };
 
 PurchaseButton.propTypes = {
+    buttonOnClick       : PropTypes.func,
     buy_info            : PropTypes.object,
     currency            : PropTypes.string,
+    icon_type           : PropTypes.string,
     index               : PropTypes.number,
     info                : PropTypes.object,
     is_contract_mode    : PropTypes.bool,
     is_disabled         : PropTypes.bool,
     is_high_low         : PropTypes.bool,
     is_loading          : PropTypes.bool,
     is_proposal_empty   : PropTypes.bool,
-    onClickPurchase     : PropTypes.func,
     purchased_states_arr: PropTypes.array,
-    setPurchaseState    : PropTypes.func,
     type                : PropTypes.string,
 };
 
-export default PurchaseButton;
+export default React.memo(PurchaseButton);
```

Comment:
I don't think it's a good one to use React memo with components which are going to re-render many times, like this one, which in every tick renders again

Reply:
React.memo doesn't make it static, but it prevents unnecessary re-renders. So this component for example was getting re-rendered every time `PurchaseFieldset` was getting re-rendered. it is a costly component, so I think it is a better approach to save a few milliseconds here and there.

Reply:
React memo is better not be used (it's a pitfall) when the component is repeatedly changing (for example like long lists), isn't purchase button going to change whenever a new tick&price comes?

Reply:
The rate of re-rendering is way more than every 2 seconds (with incoming ticks). So, rerendering `PurchaseButton` is costly (takes 15-30ms) every time we have a new render. So I don't see why we should avoid using `React.memo` when we gain performance gain from it. Isn't this an equivalent of `PureComponent`'s shallow-comparison mechanism?

Reply:
Because saving the result of a component ( = memoization) can be costly when it's done in every second, but as you had profiled it and it's working go for it

Reply:
@reza-fs I believe what @nazanin-binary is trying to say, in comparison to `PureComponent`, the PureComponent shallow compares, yes, but memoization doesn't prevent unnecessary re-renders, it just returns the same thing as before if given the same inputs, and this "saving the result of a component" (as Nazanin said) can be costly when it has to save it every 2 seconds on incoming ticks.
So even though at re-rendering takes so much time and not re-rendering takes less time, it might still be that that time is increased later on, so if after profiling you found that the cost of memoizing so often (whenever the data does indeed change) is less than the cost of re-rendering, then in that case I say we go for it too. Have you tested for this?

Reply:
I see what you guys mean, and I am not disagreeing with you. From what I remember we have saved a good amount of milliseconds with this, but as I don't have the reports for this specific component, let me try to check this one specially one more time. If it is not that major, we can ignore this one change.